File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/http_result.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class HttpResult(object):
    """
    Result from HTTP tool execution.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new HttpResult object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param status_code:
            The value to assign to the status_code property of this HttpResult.
        :type status_code: int

        :param error_code:
            The value to assign to the error_code property of this HttpResult.
        :type error_code: str

        :param message:
            The value to assign to the message property of this HttpResult.
        :type message: str

        :param is_network_error:
            The value to assign to the is_network_error property of this HttpResult.
        :type is_network_error: bool

        :param elapsed_seconds:
            The value to assign to the elapsed_seconds property of this HttpResult.
        :type elapsed_seconds: float

        :param headers:
            The value to assign to the headers property of this HttpResult.
        :type headers: dict(str, str)

        :param body:
            The value to assign to the body property of this HttpResult.
        :type body: dict(str, object)

        """
        self.swagger_types = {
            'status_code': 'int',
            'error_code': 'str',
            'message': 'str',
            'is_network_error': 'bool',
            'elapsed_seconds': 'float',
            'headers': 'dict(str, str)',
            'body': 'dict(str, object)'
        }

        self.attribute_map = {
            'status_code': 'statusCode',
            'error_code': 'errorCode',
            'message': 'message',
            'is_network_error': 'isNetworkError',
            'elapsed_seconds': 'elapsedSeconds',
            'headers': 'headers',
            'body': 'body'
        }

        self._status_code = None
        self._error_code = None
        self._message = None
        self._is_network_error = None
        self._elapsed_seconds = None
        self._headers = None
        self._body = None

    @property
    def status_code(self):
        """
        Gets the status_code of this HttpResult.
        HTTP response status code


        :return: The status_code of this HttpResult.
        :rtype: int
        """
        return self._status_code

    @status_code.setter
    def status_code(self, status_code):
        """
        Sets the status_code of this HttpResult.
        HTTP response status code


        :param status_code: The status_code of this HttpResult.
        :type: int
        """
        self._status_code = status_code

    @property
    def error_code(self):
        """
        Gets the error_code of this HttpResult.
        Tool-level error code describing HTTP test failures


        :return: The error_code of this HttpResult.
        :rtype: str
        """
        return self._error_code

    @error_code.setter
    def error_code(self, error_code):
        """
        Sets the error_code of this HttpResult.
        Tool-level error code describing HTTP test failures


        :param error_code: The error_code of this HttpResult.
        :type: str
        """
        self._error_code = error_code

    @property
    def message(self):
        """
        Gets the message of this HttpResult.
        Human-readable error message when the HTTP test fails


        :return: The message of this HttpResult.
        :rtype: str
        """
        return self._message

    @message.setter
    def message(self, message):
        """
        Sets the message of this HttpResult.
        Human-readable error message when the HTTP test fails


        :param message: The message of this HttpResult.
        :type: str
        """
        self._message = message

    @property
    def is_network_error(self):
        """
        Gets the is_network_error of this HttpResult.
        Indicates whether the failure occurred before an HTTP response was received


        :return: The is_network_error of this HttpResult.
        :rtype: bool
        """
        return self._is_network_error

    @is_network_error.setter
    def is_network_error(self, is_network_error):
        """
        Sets the is_network_error of this HttpResult.
        Indicates whether the failure occurred before an HTTP response was received


        :param is_network_error: The is_network_error of this HttpResult.
        :type: bool
        """
        self._is_network_error = is_network_error

    @property
    def elapsed_seconds(self):
        """
        Gets the elapsed_seconds of this HttpResult.
        Time elapsed before the HTTP test completed or failed


        :return: The elapsed_seconds of this HttpResult.
        :rtype: float
        """
        return self._elapsed_seconds

    @elapsed_seconds.setter
    def elapsed_seconds(self, elapsed_seconds):
        """
        Sets the elapsed_seconds of this HttpResult.
        Time elapsed before the HTTP test completed or failed


        :param elapsed_seconds: The elapsed_seconds of this HttpResult.
        :type: float
        """
        self._elapsed_seconds = elapsed_seconds

    @property
    def headers(self):
        """
        Gets the headers of this HttpResult.
        Response headers


        :return: The headers of this HttpResult.
        :rtype: dict(str, str)
        """
        return self._headers

    @headers.setter
    def headers(self, headers):
        """
        Sets the headers of this HttpResult.
        Response headers


        :param headers: The headers of this HttpResult.
        :type: dict(str, str)
        """
        self._headers = headers

    @property
    def body(self):
        """
        Gets the body of this HttpResult.
        Response body


        :return: The body of this HttpResult.
        :rtype: dict(str, object)
        """
        return self._body

    @body.setter
    def body(self, body):
        """
        Sets the body of this HttpResult.
        Response body


        :param body: The body of this HttpResult.
        :type: dict(str, object)
        """
        self._body = body

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/HttpResult.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Result from HTTP tool execution.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=HttpResult.Builder.class)

public final class HttpResult  {
    @Deprecated
    @java.beans.ConstructorProperties({"statusCode", "errorCode", "message", "isNetworkError", "elapsedSeconds", "headers", "body"})
    public HttpResult(Integer statusCode, String errorCode, String message, Boolean isNetworkError, Float elapsedSeconds, java.util.Map<String, String> headers, java.util.Map<String, Object> body) {
        super();
        this.statusCode = statusCode;
        this.errorCode = errorCode;
        this.message = message;
        this.isNetworkError = isNetworkError;
        this.elapsedSeconds = elapsedSeconds;
        this.headers = headers;
        this.body = body;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * HTTP response status code
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("statusCode")
private Integer statusCode;

        /**
         * HTTP response status code
         * @param statusCode the value to set
         * @return this builder
         **/
        

public Builder statusCode(Integer statusCode) {
    this.statusCode = statusCode;
    return this;
}
            /**
     * Tool-level error code describing HTTP test failures
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("errorCode")
private String errorCode;

        /**
         * Tool-level error code describing HTTP test failures
         * @param errorCode the value to set
         * @return this builder
         **/
        

public Builder errorCode(String errorCode) {
    this.errorCode = errorCode;
    return this;
}
            /**
     * Human-readable error message when the HTTP test fails
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("message")
private String message;

        /**
         * Human-readable error message when the HTTP test fails
         * @param message the value to set
         * @return this builder
         **/
        

public Builder message(String message) {
    this.message = message;
    return this;
}
            /**
     * Indicates whether the failure occurred before an HTTP response was received
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("isNetworkError")
private Boolean isNetworkError;

        /**
         * Indicates whether the failure occurred before an HTTP response was received
         * @param isNetworkError the value to set
         * @return this builder
         **/
        

public Builder isNetworkError(Boolean isNetworkError) {
    this.isNetworkError = isNetworkError;
    return this;
}
            /**
     * Time elapsed before the HTTP test completed or failed
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("elapsedSeconds")
private Float elapsedSeconds;

        /**
         * Time elapsed before the HTTP test completed or failed
         * @param elapsedSeconds the value to set
         * @return this builder
         **/
        

public Builder elapsedSeconds(Float elapsedSeconds) {
    this.elapsedSeconds = elapsedSeconds;
    return this;
}
            /**
     * Response headers
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("headers")
private java.util.Map<String, String> headers;

        /**
         * Response headers
         * @param headers the value to set
         * @return this builder
         **/
        

public Builder headers(java.util.Map<String, String> headers) {
    this.headers = headers;
    return this;
}
            /**
     * Response body
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("body")
private java.util.Map<String, Object> body;

        /**
         * Response body
         * @param body the value to set
         * @return this builder
         **/
        

public Builder body(java.util.Map<String, Object> body) {
    this.body = body;
    return this;
}


        public HttpResult build() {
            HttpResult model = new HttpResult(this.statusCode
                , this.errorCode
                , this.message
                , this.isNetworkError
                , this.elapsedSeconds
                , this.headers
                , this.body);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(HttpResult model) {
                this.statusCode(model.getStatusCode());
    this.errorCode(model.getErrorCode());
    this.message(model.getMessage());
    this.isNetworkError(model.getIsNetworkError());
    this.elapsedSeconds(model.getElapsedSeconds());
    this.headers(model.getHeaders());
    this.body(model.getBody());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * HTTP response status code
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("statusCode")
    private final Integer statusCode;

        /**
     * HTTP response status code
     * @return the value
     **/
    
    public Integer getStatusCode() {
        return statusCode;
    }


        /**
     * Tool-level error code describing HTTP test failures
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("errorCode")
    private final String errorCode;

        /**
     * Tool-level error code describing HTTP test failures
     * @return the value
     **/
    
    public String getErrorCode() {
        return errorCode;
    }


        /**
     * Human-readable error message when the HTTP test fails
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("message")
    private final String message;

        /**
     * Human-readable error message when the HTTP test fails
     * @return the value
     **/
    
    public String getMessage() {
        return message;
    }


        /**
     * Indicates whether the failure occurred before an HTTP response was received
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("isNetworkError")
    private final Boolean isNetworkError;

        /**
     * Indicates whether the failure occurred before an HTTP response was received
     * @return the value
     **/
    
    public Boolean getIsNetworkError() {
        return isNetworkError;
    }


        /**
     * Time elapsed before the HTTP test completed or failed
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("elapsedSeconds")
    private final Float elapsedSeconds;

        /**
     * Time elapsed before the HTTP test completed or failed
     * @return the value
     **/
    
    public Float getElapsedSeconds() {
        return elapsedSeconds;
    }


        /**
     * Response headers
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("headers")
    private final java.util.Map<String, String> headers;

        /**
     * Response headers
     * @return the value
     **/
    
    public java.util.Map<String, String> getHeaders() {
        return headers;
    }


        /**
     * Response body
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("body")
    private final java.util.Map<String, Object> body;

        /**
     * Response body
     * @return the value
     **/
    
    public java.util.Map<String, Object> getBody() {
        return body;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("HttpResult(");
        sb.append("statusCode=").append(String.valueOf(this.statusCode));
        sb.append(", errorCode=").append(String.valueOf(this.errorCode));
        sb.append(", message=").append(String.valueOf(this.message));
        sb.append(", isNetworkError=").append(String.valueOf(this.isNetworkError));
        sb.append(", elapsedSeconds=").append(String.valueOf(this.elapsedSeconds));
        sb.append(", headers=").append(String.valueOf(this.headers));
        sb.append(", body=").append(String.valueOf(this.body));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof HttpResult)) {
            return false;
        }

        HttpResult other = (HttpResult) o;
        return java.util.Objects.equals(this.statusCode, other.statusCode) &&
            java.util.Objects.equals(this.errorCode, other.errorCode) &&
            java.util.Objects.equals(this.message, other.message) &&
            java.util.Objects.equals(this.isNetworkError, other.isNetworkError) &&
            java.util.Objects.equals(this.elapsedSeconds, other.elapsedSeconds) &&
            java.util.Objects.equals(this.headers, other.headers) &&
            java.util.Objects.equals(this.body, other.body);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.statusCode == null ? 43 : this.statusCode.hashCode());
        result = (result * PRIME) + (this.errorCode == null ? 43 : this.errorCode.hashCode());
        result = (result * PRIME) + (this.message == null ? 43 : this.message.hashCode());
        result = (result * PRIME) + (this.isNetworkError == null ? 43 : this.isNetworkError.hashCode());
        result = (result * PRIME) + (this.elapsedSeconds == null ? 43 : this.elapsedSeconds.hashCode());
        result = (result * PRIME) + (this.headers == null ? 43 : this.headers.hashCode());
        result = (result * PRIME) + (this.body == null ? 43 : this.body.hashCode());
        return result;
    }


}
